staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -205,7 +204,7 @@ func newTimeBucketWatchers(clock clock.Clock, bookmarkFrequency time.Duration) *
 
 // adds a watcher to the bucket, if the deadline is before the start, it will be
 // added to the first one.
-func (t *watcherBookmarkTimeBuckets) addWatcher(w *cacheWatcher) bool {
+func (t *watcherBookmarkTimeBuckets) addWatcherThreadUnsafe(w *cacheWatcher) bool {
```

Comment:
Can you also in this line:
https://github.com/kubernetes/kubernetes/blob/e4ee8fded488ee400f03862f83b3d46a3ec8a4c5/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L326

(where `bookmarkWatchers` field is defined)
add an additional comment, that this field when accessed should be protected by the Cacher.lock itself.

Reply:
done